Habitica/src/main/java/com/habitrpg/android/habitica/ui/views/subscriptions/SubscriptionDetailsView.kt

```diff
@@ -96,8 +105,15 @@ class SubscriptionDetailsView : LinearLayout {
         } else {
             binding.monthsSubscribedTextView.text = resources.getString(R.string.x_months, plan.consecutive?.count ?: 0)
         }
+
         binding.gemCapTextView.text = plan.totalNumberOfGems.toString()
-        binding.currentHourglassesTextView.text = plan.consecutive?.trinkets.toString()
+
+        if (plan.consecutive?.count != null) {

```

Comment:
Can you move this calculation out of the view and into either a separate class or the SubscriptionPlan class? And then add a unit test to it, so that we can make it check a few cases and be sure it works as expected.

Reply:
For the actual tests, we can also go over the test cases together to make sure we catch as many cases as possible.

Reply:
For sure I can do that